integrations/slack/client.go

```diff
@@ -34,230 +37,257 @@ var desc = kittehs.Must1(sdktypes.StrictIntegrationFromProto(&sdktypes.Integrati
 	},
 }))
 
-func New(vars sdkservices.Vars) sdkservices.Integration {
-	authAPI := auth.API{Vars: vars}
-	bookmarksAPI := bookmarks.API{Vars: vars}
-	botsAPI := bots.API{Vars: vars}
-	chatAPI := chat.API{Vars: vars}
-	conversationsAPI := conversations.API{Vars: vars}
-	reactionsAPI := reactions.API{Vars: vars}
-	usersAPI := users.API{Vars: vars}
-
+func New(vs sdkservices.Vars) sdkservices.Integration {
 	return sdkintegrations.NewIntegration(
 		desc,
-		sdkmodule.New(
-			// Auth.
-			sdkmodule.ExportFunction(
-				"auth_test",
-				authAPI.Test,
-				sdkmodule.WithFuncDoc("https://api.slack.com/methods/auth.test"),
-			),
+		sdkmodule.New(funcs(vs)...),
+		connStatus(vs),
+		sdkintegrations.WithConnectionConfigFromVars(vs),
+	)
+}
 
-			// Bookmarks.
-			sdkmodule.ExportFunction(
-				"bookmarks_add",
-				bookmarksAPI.Add,
-				sdkmodule.WithFuncDoc("https://api.slack.com/methods/bookmarks.add"),
-				sdkmodule.WithArgs("channel_id", "title" /* , "type" */, "link?", "emoji?", "entity_id?", "parent_id?"),
-			),
-			sdkmodule.ExportFunction(
-				"bookmarks_edit",
-				bookmarksAPI.Edit,
-				sdkmodule.WithFuncDoc("https://api.slack.com/methods/bookmarks.edit"),
-				sdkmodule.WithArgs("bookmark_id", "channel_id", "emoji?", "link?", "title?"),
-			),
-			sdkmodule.ExportFunction(
-				"bookmarks_list",
-				bookmarksAPI.List,
-				sdkmodule.WithFuncDoc("https://api.slack.com/methods/bookmarks.list"),
-				sdkmodule.WithArgs("channel_id"),
-			),
-			sdkmodule.ExportFunction(
-				"bookmarks_remove",
-				bookmarksAPI.Remove,
-				sdkmodule.WithFuncDoc("https://api.slack.com/methods/bookmarks.remove"),
-				sdkmodule.WithArgs("bookmark_id", "channel_id", "quip_section_id?"),
-			),
+// connStatus is an optional connection status check provided by the
+// integration to AutoKitteh. The possible results are "init required"
+// (the connection is not usable yet) and "using X" (where "X" is the
+// authentication method: an OAuth v2 app, or a Socket Mode app).
```

Comment:
There should also be an `error` status to differentiate from regular error which means `I can't get the connection status`. The connection error status can contain reason, say invalid credentials.

Reply:
In theory I agree, and it will be done, but for now we don't have a check for stateful connections.